node/node.go

```diff
@@ -510,12 +514,26 @@ func makeSeedNode(cfg *config.Config,
 // OnStart starts the Node. It implements service.Service.
 func (n *nodeImpl) OnStart(ctx context.Context) error {
 	if n.config.RPC.PprofListenAddress != "" {
-		// this service is not cleaned up (I believe that we'd
-		// need to have another thread and a potentially a
-		// context to get this functionality.)
 		go func() {
+			sig := make(chan struct{})
+
 			n.Logger.Info("Starting pprof server", "laddr", n.config.RPC.PprofListenAddress)
-			n.Logger.Error("pprof server error", "err", http.ListenAndServe(n.config.RPC.PprofListenAddress, nil))
+			srv := &http.Server{Addr: n.config.RPC.PprofListenAddress, Handler: nil}
+
+			go func() {
```

Comment:
I think this could potentially be easier to read if we wrap the ctx passed to OnStart, and de-nest the goroutines.

```go
ctx, cancel := context.WithCancel(ctx)
srv := &http.Server{ /* blah blah */}

go func() {
   <-ctx.Done()
   sctx, scancel := context.WithTimeout(context.Background(), time.Second)
   defer scancel()
   srv.Shutdown(sctx)
}()

go func() {
   if err := srv.ListenAndServe(); err != nil {
      // log
      cancel()
   }
}()
```

YMMV.